### DISCUSSION TITLE
Very slow with big for loop

### DISCUSSION THREAD

**Discussion by ghost** [lcn2/calc#120] [General]
`for(i=1;i<N;i++)`

If `N = 2147483647`, it is too slow to complete, I have to cancel it. With small N, it is quite fast.

**Comment by pmetzger** [comment]
You don't specify what is in the loop, but if you're doing anything that takes a moderate amount of time two billion times, it is going to take a while. Say something takes just a millisecond; do it fifty times and you won't notice. Do it two billion times and it's 40 minutes to complete.

**Comment by lcn2** [comment]
Calc is an interpreted language, not compiled.

On Apple silicon M1 Max:

```calls
; N=2147483647; start=runtime(); for(i=1;i<N;i++); print runtime()-start;
166.517041
```

165.5 seconds.

**Comment by lcn2** [comment]
We have added it as a TODO to consider a JIT for calc in version 3.

See [comment 1755671627](https://github.com/lcn2/calc/issues/103#issuecomment-1755671627).

We will close this discussion here and continue it in issue #103 if you wish to post follow up matters.

Thanks for raising this issue.

**Comment by pmetzger** [comment]
I note that the original requester posted with a brand new account and then deleted that account.
